daemon/config.go

```diff
@@ -57,4 +67,6 @@ func (config *Config) InstallCommonFlags(cmd *flag.FlagSet, usageFn func(string)
 	cmd.Var(opts.NewListOptsRef(&config.Labels, opts.ValidateLabel), []string{"-label"}, usageFn("Set key=value labels to the daemon"))
 	cmd.StringVar(&config.LogConfig.Type, []string{"-log-driver"}, "json-file", usageFn("Default driver for container logs"))
 	cmd.Var(opts.NewMapOpts(config.LogConfig.Config, nil), []string{"-log-opt"}, usageFn("Set log driver options"))
+	cmd.StringVar(&config.DiscoveryAddress, []string{"-discovery-address"}, "", usageFn("Address of the daemon instance to advertise"))
```

Comment:
@icecrime  Should we validate the `-discovery-address`?  Since this is store in discovery backend for discovered by others,
and the node ip is the key, we should ensure it has a unique format.  For example if we start `-discovery-address=http://9.81.1.185:2376`,
it will be like this in etcd
<pre><code>{
  "node": {
    "createdIndex": 1459,
    "modifiedIndex": 1459,
    "nodes": [
      {
        "createdIndex": 1459,
        "modifiedIndex": 1459,
        "dir": true,
        "key": "/docker/nodes/http:"
      }
    ],
    "dir": true,
    "key": "/docker/nodes"
  },
  "action": "get"</code></pre>
the `http` become the key of the node
it supposed to like this
<pre><code>"createdIndex": 1479,
        "modifiedIndex": 1479,
        "ttl": 57,
        "expiration": "2015-09-19T07:06:18.144476351Z",
        "value": "9.81.1.185:2376",
        "key": "/docker/nodes/9.81.1.185:2376"</code></pre>
So I think we should make the user use the right format of `-discovery-address`